### ISSUE TITLE
Can't return from a non-function

### ISSUE THREAD

**Issue by appsforartists** [dominictarr/my-local-ip#3]
Both Node and Babel seems to be choking on this return, when my-local-ip is used as a module:

``` javascript
if (!module.parent)
  return console.log(require("./get_ip")())
```

```
SyntaxError: node_modules/my-local-ip/index.js: 'return' outside of function (18:2)
```

**Comment by dominictarr** [comment]
That works in node because node wraps every module in a closure, it's part of how the module system works. if it isn't working in node, what version of node does it fail in?

**Comment by appsforartists** [comment]
I think I copied and pasted the code into a REPL after Babel kept choking on it.  That's what I get for doing this stuff at 2 AM.  :confused: 

Nevertheless, it is making Babel choke.  What's the point of returning undefined in the last line of the module?  `module.exports` is already set to the function.

**Comment by dominictarr** [comment]
I'm not trying to maliciously break babel, but since this works in node and node is a defacto standard, prehaps this should be considered a babel bug?

I havn't used babel, but my understanding is that it compiles es6 to es5? (is that correct?)
es6 has modules right? which become node style require? so therefore it is tied to node and I think this ought to work if it works in node.

**Comment by appsforartists** [comment]
It is indeed based on ES6.  If you can't return from outside a function in traditional JS, I don't know why you'd be able to return from a module.  (The only official spec I can find is too dense for me to parse right now on this issue.)

More importantly, what's the significance of `return console.log()` in your code?  Your export is the `myIp` function.  What are you trying to accomplish with the `return` statement?

**Comment by dominictarr** [comment]
maybe it's just there because something happened after that, and I didn't remove it after changing that or something...

anyway, lets make a deal - you post an issue about this on babel, and I'll remove it here.

**Comment by appsforartists** [comment]
Thanks.

I'm not convinced of the utility of changing this in Babel.  What would it mean to return from a module definition?  I'm not even sure that's valid in the ES6 spec.

**Comment by dominictarr** [comment]
No, but this file is not es6, it's written in node. node existed before babel did, and this has always worked. If babel is gonna work with node _properly_ it needs to support this. or else I bet you this will happen again and you'll have to make all these prs

**Comment by appsforartists** [comment]
Perhaps.

I'm still trying to understand the value (or even the meaning) in returning from a module definition.  It seems more like a hack/edge case than anything that was ever intentional.

**Comment by dominictarr** [comment]
okay well, i use it quite often use it to disable tests that i don't need to run.

``` js
test1()

test2()

return

test3() //disabled...
```

node modules are really like 

``` js
function (require, module, exports, __dirname, __filename) {

  test1()

  test2()

  return

  test3() //disabled...

}
```

so when it's a node module it's never actually at the top. this was a problem with browserify too, but they fixed it by wrapping each file in a closure before they parsed.
